Task 3 forms directives pipes

OlympiqueB opened the pull request:
Almost half of the Autocode tests fail, but the components seem to be working.
Some of them were not completely clear to me, but I've tried to implement them based on the task instructions.

OlympiqueB committed:
email directive added

OlympiqueB committed:
added click event to custom button

OlympiqueB committed:
login-form done

OlympiqueB committed:
error style modified to fit template-driven form

OlympiqueB committed:
removed unused imports

OlympiqueB committed:
registration-form done

OlympiqueB committed:
added email validation

OlympiqueB committed:
numberOnly directive created

OlympiqueB committed:
work on course-form component

OlympiqueB committed:
Merge remote-tracking branch 'origin/main' into task-3-forms-directives-pipes

OlympiqueB committed:
modified to show 0:00 even when duration is undefined

OlympiqueB committed:
modified directive to correctly handle writing multiple leading 0s

OlympiqueB committed:
fixed duration value handling

OlympiqueB committed:
added buttonLabelService

OlympiqueB committed:
author model added

OlympiqueB committed:
installed uuid

OlympiqueB committed:
added more labels

OlympiqueB committed:
removed clg

OlympiqueB committed:
course form done

OlympiqueB committed:
simplified onsubmit, changed styling process

OlympiqueB committed:
invalid styling added to newAuthor

OlympiqueB committed:
moved course-list to courses component

OlympiqueB committed:
moved courses-list to course component

OlympiqueB committed:
search label added

OlympiqueB committed:
search component set up

OlympiqueB committed:
rearranged

OlympiqueB committed:
removed comment

OlympiqueB committed:
adjusted error messages

TamasViczian commented on src/app/shared/components/course-form/course-form.component.ts:
```diff
@@ -1,19 +1,142 @@
-export class CourseFormComponent {
-  constructor(public fb: FormBuilder, public library: FaIconLibrary) {
+export class CourseFormComponent implements OnInit {
+  constructor(
```
Hmmm, I think these constructor params can be assigned private visibility

TamasViczian commented on src/app/shared/components/course-form/course-form.component.ts:
```diff
@@ -1,19 +1,142 @@
+        authors: this.authors.value,
+      };
+
+      console.log(newCourse);
```
you can delete

TamasViczian commented on src/app/shared/components/course-form/course-form.component.ts:
```diff
@@ -1,19 +1,142 @@
+        name: this.newAuthor.value,
+      };
+      this.fullAuthorArray.push(newAuthor);
+      console.log(newAuthor);
```
delete

TamasViczian commented on src/app/shared/components/course-form/course-form.component.html:
A few observations on functionality:

Once I click the create course button without the form filled out, here is the UI update. I guess it is not compulsory to enter an author name - surely it is ok to select some author(s) from the list. But 'Course Authors' does not give any feedback on no authors added to the course.
![image](https://github.com/user-attachments/assets/19961aad-2e2d-4af9-bb32-da0f2c4d707d)

I think whitespace character is allowed to add in the Author create field.
![image](https://github.com/user-attachments/assets/21fa1097-e8ad-461a-9d2d-b299f5390ea5)

TamasViczian commented on src/app/shared/components/login-form/login-form.component.html:
I see that the once I hit login button, the console logs the credentials and the UI is updated like below.
- probably we don't want to log the user password anywhere as that can be a major security threat.
- form can be reset, instead of invalid status
![image](https://github.com/user-attachments/assets/e468a5ca-a175-461a-82bf-12a1ad7c01a3)

TamasViczian commented on src/app/shared/components/login-form/login-form.component.ts:
```diff
@@ -1,12 +1,31 @@
+        email: this.loginForm.value.email,
+        password: this.loginForm.value.password,
+      };
+      console.log(loginUser);
```
delete

TamasViczian commented on src/app/shared/components/registration-form/registration-form.component.ts:
```diff
@@ -1,12 +1,56 @@
+        email: this.email,
+        password: this.password,
+      };
+      console.log(newUser);
```
delete?

TamasViczian reviewed (changes requested):
Really nice job! There are a few updates you might want to consider.
As I see you are getting along with the task pretty good, I suggest you implement the extra task with password show/hide. Thanks

OlympiqueB committed:
removed console.logs

OlympiqueB committed:
constructor params set to private

OlympiqueB committed:
added message if course author list is empty

OlympiqueB committed:
reset author form so it is not styled invalid

OlympiqueB committed:
adjusted validation pattern, styling and message

OlympiqueB committed:
removed else

OlympiqueB committed:
added eye icons to label service

OlympiqueB committed:
togglePassword directive added

OlympiqueB committed:
added togglePassword directive to login form

OlympiqueB commented:
Hi,

Thanks for the review and the comments.
I think I've addressed the issues.

I also added the directive for password toggle, which I applied to the login-form component. The styling and the icon implementation might be a bit messy. It would probably make more sense to create a separate icon service for handling this, but this was made in a hurry, just to test and show that the directive is working.

TamasViczian commented on src/app/shared/components/login-form/login-form.component.ts:
```diff
@@ -10,22 +13,32 @@ export class LoginFormComponent {
+    public buttonLabelService: ButtonLabelService,
+    private library: FaIconLibrary
+  ) {
+    {
```
Double curly braces?

TamasViczian reviewed (approved):
Good, approved. Just delete the duplicated curly braces on LoginFormComponent constructor.

OlympiqueB committed:
removed unnecessary curly braces
